### PR TITLE
awss3exporter: Add Protocol Buf storage format

### PR DESCRIPTION
**Pull request by adcharre**
**Description:** Add ProtoBuf output format to the AWS S3 exporter

**Link to tracking Issue:** #30681

**Testing:** Additional unit tests have been added for the new format type and I've also exported data to an S3 bucket in Protobuf format and read it back to confirm that it's working end to end.

**Documentation:** Added details of the new marshaler format to the README.md

### CODE REVIEW

File: .chloggen/awss3-add-protobuf-marshaller.yaml [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+change_type: enhancement
+
+# The name of the component, or a single word describing the area of concern, (e.g. filelogreceiver)
+component: awss3exporter
+
+# A brief description of the change.  Surround your text with quotes ("") if it needs to start with a backtick (`).
+note: "Add the ability to export trace/log/metrics in OTLP ProtoBuf format."
+
+# Mandatory: One or more tracking issues related to the change. You can use the PR number here if no issue exists.
+issues: []
```
Comment (atoulme) [inline .chloggen/awss3-add-protobuf-marshaller.yaml:13]: you need an issue number here. You can enter this PR number.
```suggestion
issues: [30682]
```

File: exporter/awss3exporter/marshaler_test.go [hunk 2]
```diff
@@ -18,6 +18,12 @@ func TestMarshaler(t *testing.T) {
 		require.NotNil(t, m)
 		assert.Equal(t, m.format(), "json")
 	}
+	{
+		m, err := newMarshaler("otlp_proto", zap.NewNop())
+		assert.NoError(t, err)
+		require.NotNil(t, m)
+		assert.Equal(t, m.format(), "binpb")
```
Comment (atoulme) [inline exporter/awss3exporter/marshaler_test.go:25]: Is `binpb` a good term here?

Comment (adcharre) [reply exporter/awss3exporter/marshaler_test.go:25]: I did have to look up what a binary protocol buffer file extension should be, according to the Protocol Buf site:
https://protobuf.dev/programming-guides/techniques/#suffixes

Comment (atoulme) [reply exporter/awss3exporter/marshaler_test.go:25]: Nice, thanks for the link! :)

**Review by atoulme — APPROVED** [review]
Looks tight, LGTM

**Comment by atoulme** [comment]
Any @open-telemetry/collector-contrib-maintainer willing to give their blessing and merge this PR?

**Review by TylerHelmuth — COMMENTED** [review]
I dont see any tests that prove these changes do what they claim to do, were you able to test this locally an confirm the proto marshalling works as expected?

**Comment by adcharre** [comment]
> I dont see any tests that prove these changes do what they claim to do, were you able to test this locally an confirm the proto marshalling works as expected?

I was indeed able to confirm that the existing Protobuf Marshaler does generate a byte stream that can be unmarshaled when read back from S3.

**Review by TylerHelmuth — APPROVED** [review]

**Pull request merged**